<!DOCTYPE HTML PUBLIC '-//W3C//DTD HTML 4.01 Transitional//EN'>
<html>

<head>
<meta http-equiv='Content-Type' content='text/html; charset=utf-8'>
<meta http-equiv='Content-Style-Type' content='text/css'>
<meta name='author' content='Yuki Naito'>
<title>CRISPRdirect help</title>
<style type='text/css'>
<!--
	*     { font-family:verdana,arial,helvetica,sans-serif }
	.mono { font-family:monaco,courier,monospace }
	.pam  {
		font-family:inherit;
		border:1px solid;
		margin-left:1px;
		margin-right:1px;
		padding-left:1px;
		padding-right:1px;
	}
	p,div,ol,ul,h5 { font-size:9pt }
	table,textarea { font-size:8pt }
	p,ol,ul { width:90% }
	a       { text-decoration:none; color:#003399 }
	a:hover { text-decoration:none; color:maroon  }
	.b      { text-decoration:none; color:black   }
	.s      { text-decoration:none; color:gray    }
	form  { margin:10px 0px }
	table#apiman { white-space:nowrap; border-collapse:collapse }
	table#apiman th {
		background-color:#E6E6FA;
		border:solid 1px #8C93BA;
		padding:4px
	}
	table#apiman td {
		border:solid 1px #8C93BA;
		padding:4px
	}
	em { background-color:#99FF99; font-style:normal; font-weight:bold }
	.v { background-color:#E6E6FA }
	.o { background-color:#FFF0E0 }
	.g { background-color:#E6FAF2 }
	.h { background-color:#99FF99 }

	/* zurui_button from http://css3button.net/ */
	.zbutton {
		text-decoration:none;
		outline:none;
		padding:2px 8px;
		border:1px solid #606060;
		-webkit-border-radius:9px;
		-moz-border-radius:9px;
		-ms-border-radius:9px;
		border-radius:9px;
		background:-webkit-gradient(linear, left top, left bottom, from(#E5EEFF), to(#BFD4FF));
		background:-webkit-linear-gradient(top, #E5EEFF, #BFD4FF);
		background:-moz-linear-gradient(top, #E5EEFF, #BFD4FF);
		background:-ms-linear-gradient(top, #E5EEFF, #BFD4FF);
		background:-o-linear-gradient(top, #E5EEFF, #BFD4FF);
		background:linear-gradient(top, #E5EEFF, #BFD4FF);
		background-color:#E5EEFF;
		text-shadow:
			0px -1px 0px rgba(0,0,0,0.2),
			0px  1px 0px rgba(255,255,255,0.5);
	}
	.zbutton:hover {
		background:-webkit-gradient(linear, left top, left bottom, from(#CCDEFF), to(#80AAFF));
		background:-webkit-linear-gradient(top, #CCDEFF, #80AAFF);
		background:-moz-linear-gradient(top, #CCDEFF, #80AAFF);
		background:-ms-linear-gradient(top, #CCDEFF, #80AAFF);
		background:-o-linear-gradient(top, #CCDEFF, #80AAFF);
		background:linear-gradient(top, #CCDEFF, #80AAFF);
		background-color:#CCDEFF;
	}
	.zbutton:active {
		color:white;
		background:-webkit-gradient(linear, left top, left bottom, from(#8296BD), to(#2A5EBC));
		background:-webkit-linear-gradient(top, #8296BD, #2A5EBC);
		background:-moz-linear-gradient(top, #8296BD, #2A5EBC);
		background:-ms-linear-gradient(top, #8296BD, #2A5EBC);
		background:-o-linear-gradient(top, #8296BD, #2A5EBC);
		background:linear-gradient(top, #8296BD, #2A5EBC);
		background-color:#8296BD;
	}
-->
</style>
</head>

<body>

<div>
<a href='..'><img src='CRISPRdirect.png' alt='CRISPRdirect' width=200 height=105 border=0></a>
<span style='vertical-align:4.3ex'>&mdash; Rational design of CRISPR/Cas target.</span>
</div>

<hr style='border-top:3px solid #2655A9'><!-- ________________________________________ -->


<h4>About CRISPRdirect</h4>

<p>CRISPRdirect is a web server for selecting rational CRISPR/Cas targets from an input sequence.<br>
	All services of the CRISPRdirect web server are provided free of charge to all users.</p>


<h4>Tutorial video</h4>

<a target='_blank' href='http://togotv.dbcls.jp/en/?p=433'>
<img src='togotv.png' alt='CRISPRdirect: designing CRISPR/Cas guide RNA sequence'
	height=270 width=480 border=0></a>
<p class=s>Acknowledgement: We thank Ms. Junko Morita for making the tutorial video on
	<a target='_blank' href='http://togotv.dbcls.jp/en/'>TogoTV</a>.</p>


<h4>Example</h4>

<h5>Example 1 | Selecting target sites for human B melanoma antigen (BAGE) gene</h5>

<ol>
	<li>Enter the accession number for human BAGE gene (NM_001187).
	<li>Click 'retrieve sequence' to get nucleotide sequence from GenBank.

	<form name=getsequence method=GET action='../' target='_blank'>
	<span style='border:dotted 1px #8C93BA; padding:7px'>
		Show example:
		<input type=hidden name=accession value='NM_001187'>
		<input type=submit value='retrieve sequence'>
	</span>
	</form>

	<li>Or you can directly paste a nucleotide sequence.<br>
		FASTA format or a plain nucleotide sequence up to 10 kbp is accepted.
	<li>Or you can upload a sequence file, in plain text format. (not MS Word .doc file!)<br>
		FASTA format or a plain nucleotide sequence up to 10 kbp is accepted.
	<li>PAM sequence requirement (default: NGG).<br>
		An arbitrary 3 nt sequence using IUB codes (N, R, Y, ...) can be specified.
	<li>Select 'Human genome, GRCh37/hg19 (Feb, 2009)' to check for off-targets.
	<li>Click 'design'.

	<form name=crisprdirect method=POST action='../' target='_blank'>
	<span style='border:dotted 1px #8C93BA; padding:7px'>
		Show example:
		<input type=hidden name=accession value='NM_001187'>
		<input type=hidden name=userseq value='>gi|4557346|ref|NM_001187.1| Homo sapiens B melanoma antigen (BAGE), mRNA
CGCCAATTTAGGGTCTCCGGTATCTCCCGCTGAGCTGCTCTGTTCCCGGCTTAGAGGACCAGGAGAAGGG
GGAGCTGGAGGCTGGAGCCTGTAACACCGTGGCTCGTCTCACTCTGGATGGTGGTGGCAACAGAGATGGC
AGCGCAGCTGGAGTGTTAGGAGGGCGGCCTGAGCGGTAGGAGTGGGGCTGGAGCAGTAAGATGGCGGCCA
GAGCGGTTTTTCTGGCATTGTCTGCCCAGCTGCTCCAAGCCAGGCTGATGAAGGAGGAGTCCCCTGTGGT
GAGCTGGAGGTTGGAGCCTGAAGACGGCACAGCTCTGTGCTTCATCTTCTGAGGTTGTGGCAGCCACGGT
GATGGAGACGGCAGCTCAACAGGAGCAATAGGAGGAGATGGAGTTTCACTGTGTCAGCCAGGATGGTCTC
GATCTCCTGACCTCGTGATCCGCCCGCCTTGGCCTTCCAAAGTGCCGAGATTACAGCGATGTGCATTTTG
TAAGCACTTTGGAGCCACTATCAAATGCTGTGAAGAGAAATGTACCCAGATGTATCATTATCCTTGTGCT
GCAGGAGCCGGCTCCTTTCAGGATTTCAGTCACATCTTCCTGCTTTGTCCAGAACACATTGACCAAGCTC
CTGAAAGATGTAAGTTTACTACGCATAGACTTTTAAACTTCAACCAATGTATTTACTGAAAATAACAAAT
GTTGTAAATTCCCTGAGTGTTATTCTACTTGTATTAAAAGGTAATAATACATAATCATTAAAATCTGAGG
GATCATTGCCAGAGATTGTTGGGGAGGGAAATGTTATCAACGGTTTCATTGAAATTAAATCCAAAAAGTT
ATTTCCTCAGAAAAATCAAATAAAGTTTGCATGTTTTTTATTCTTAAAACATTTTAAAAACCACTGTAGA
ATGATGTAAATAGGGACTGTGCAGTATTTCTGACATATACTATAAAATTATTAAAAAGTCAATCAGTATT
CAACATCTTTTACACTAAAAAGCC
'>
		<input type=hidden name=db value=hg19>
		<input type=submit class=zbutton value='   design   '>
	</span>
	</form>
</ol>

<img src='CRISPR_BAGE_fasta.png' width=557 height=458 border=0>

<h5>Results</h5>

<ol type=A>
	<li>Target positions.
	<li>Target sequences, 20mer + 3mer PAM (total 23mer).
	<li>GC content of the target 20mer.
	<li>Calculated Tm of the target 20mer.
	<li>Presence or absence of TTTT (four consecutive Tâ€™s that cause pol III termination)
		in the target 20mer. Avoid TTTT in gRNA vectors with pol III promoter.
	<li>Off-target search results against human genome. The number of target sites with perfect match
		is shown. Note that the number displayed here includes both on-target and off-target sites.
		Smaller number (but not zero) is better for these columns to avoid off-target editing.
	<ul>
		<li>20mer+PAM: The target sequence (20mer) adjacent to the PAM is searched against the genome.
			One ('1') in this column usually indicates that the sequence has only one perfect match
			with the intended target (e.g., BAGE locus in the genome).
			Zero ('0') in this column means that the sequence has no match in the genomic sequence;
			such sequences may possibly span over exon-exon junctions, so their use should be avoided.
		<li>12mer+PAM: The 12mer of 3&prime; region of the target sequence adjacent to the PAM is searched
			against the genome. The region contains critical residues determining target specificity.
		<li>8mer+PAM: The 8mer of 3&prime; region of the target sequence adjacent to the PAM is searched
			against the genome. The region contains the most critical residues determining target specificity.
	</ul>
</ol>

<img src='CRISPR_BAGE_result1.png' width=557 height=430 border=0>

<ol type=A start=7>
	<li>Only one match in 20mer+PAM and 12mer+PAM search. These targets are highly specific.<br>
		<font color=brown>&rarr; Recommended for CRISPR/Cas target.</font>
		Target positions are highlighted with green (e.g., <em>163 - 185</em>).
	<li>No match in 20mer+PAM search. Possibly the sequence spans over exon-exon junction,
		so avoid using these.<br>
		<font color=brown>&rarr; Not recommended for CRISPR/Cas target.</font>
	<li>Very high number of off-target hits. Avoid using these sequences.<br>
		<font color=brown>&rarr; Not recommended for CRISPR/Cas target.</font>
</ol>

<img src='CRISPR_BAGE_result2.png' width=557 height=356 border=0>

<h5>Graphical view</h5>

<p>A graphical view of target sites demonstrates the position and orientation of each site.</p>

<img src='CRISPR_BAGE_graview.png' width=557 height=324 border=0>

<h5>Saving and exporting the results</h5>

<ol type=A start=10>
	<li>Tab-delimited text can be displayed or downloaded as a file.
	<li>JSON format output is also available.
	<li>Tab-delimited text, ready-to-use for copy-pasting into Excel or text editors, etc.
</ol>

<img src='CRISPR_BAGE_tsv.png' width=557 height=462 border=0>

<p>The result page can also be saved as HTML file and opened by a web browser again.</p>

<img src='savehtml.png' width=212 height=172 border=0>

<h5>Example 2 | Analyzing an existing gRNA target sequence</h5>

<ol>
	<li>Any 20 nt sequence followed by PAM (i.e. NGG) can be simply submitted as an input.<br>
		An example for verifying the target sequence 'ACCTCTCTGAGGCTGCCAACCGG':
</ol>

<img src='CRISPRverify.png' width=557 height=470 border=0>


<h4>Methods</h4>

<h5>Specificity check</h5>

<p>The number of target sites in the genome (including both on-target and off-target sites)
	is displayed using <a target='_blank' href='http://www.cbcb.umd.edu/software/jellyfish/'>
	Jellyfish</a> k-mer counting software. A detailed list of off-target candidates are
	displayed using <a target='_blank' href='http://GGGenome.dbcls.jp/'>GGGenome</a>
	REST API. GGGenome quickly searches short nucleotide sequences allowing mismatches and
	gaps utilizing suffix arrays and inverse suffix links indexed on solid state drive (SSD). </p>

<p>Specificity check can be performed for following species:</p>

<ul>
	<li><b>Human (<i>Homo sapiens</i>) genome</b>, GRCh37/hg19 (Feb, 2009)<br>
		<a target='_blank' href='ftp://hgdownload.soe.ucsc.edu/goldenPath/hg19/bigZips/chromFa.tar.gz'>
		ftp://hgdownload.soe.ucsc.edu/goldenPath/hg19/bigZips/chromFa.tar.gz</a>
	<li><b>Mouse (<i>Mus musculus</i>) genome</b>, GRCm38/mm10 (Dec, 2011)<br>
		<a target='_blank' href='ftp://hgdownload.soe.ucsc.edu/goldenPath/mm10/bigZips/chromFa.tar.gz'>
		ftp://hgdownload.soe.ucsc.edu/goldenPath/mm10/bigZips/chromFa.tar.gz</a>
	<li><b>Rat (<i>Rattus norvegicus</i>) genome</b>, RGSC 5.0/rn5 (Mar, 2012)<br>
		<a target='_blank' href='ftp://hgdownload.soe.ucsc.edu/goldenPath/rn5/bigZips/chromFa.tar.gz'>
		ftp://hgdownload.soe.ucsc.edu/goldenPath/rn5/bigZips/chromFa.tar.gz</a>
	<li><b>Marmoset (<i>Callithrix jacchus</i>) genome</b>, WUGSC 3.2/calJac3 (Mar, 2009)<br>
		<a target='_blank' href='ftp://hgdownload.soe.ucsc.edu/goldenPath/calJac3/bigZips/calJac3.fa.gz'>
		ftp://hgdownload.soe.ucsc.edu/goldenPath/calJac3/bigZips/calJac3.fa.gz</a>
	<li><b>Pig (<i>Sus scrofa</i>) genome</b>, SGSC Sscrofa10.2/susScr3 (Aug, 2011)<br>
		<a target='_blank' href='ftp://hgdownload.soe.ucsc.edu/goldenPath/susScr3/bigZips/susScr3.fa.gz'>
		ftp://hgdownload.soe.ucsc.edu/goldenPath/susScr3/bigZips/susScr3.fa.gz</a>
	<li><b>Chicken (<i>Gallus gallus</i>) genome</b>, ICGSC Gallus_gallus-4.0/galGal4 (Nov, 2011)<br>
		<a target='_blank' href='ftp://hgdownload.soe.ucsc.edu/goldenPath/galGal4/bigZips/galGal4.fa.gz'>
		ftp://hgdownload.soe.ucsc.edu/goldenPath/galGal4/bigZips/galGal4.fa.gz</a>
	<li><b>Frog (<i>Xenopus tropicalis</i>) genome</b>, JGI 4.2/xenTro3 (Nov, 2009)<br>
		<a target='_blank' href='ftp://hgdownload.soe.ucsc.edu/goldenPath/xenTro3/bigZips/xenTro3.fa.gz'>
		ftp://hgdownload.soe.ucsc.edu/goldenPath/xenTro3/bigZips/xenTro3.fa.gz</a>
	<li><b>Frog (<i>Xenopus laevis</i>) genome</b>, JGI 7.1/Xenla7 (Dec, 2013)<br>
		<a target='_blank' href='ftp://ftp.xenbase.org/pub/Genomics/JGI/Xenla7.1/Xenla_7.1_JGI_scaffolds_on_Xenbase_GBrowse.tgz'>
		ftp://ftp.xenbase.org/pub/Genomics/JGI/Xenla7.1/Xenla_7.1_JGI_scaffolds_on_Xenbase_GBrowse.tgz</a>
	<li><b>Zebrafish (<i>Danio rerio</i>) genome</b>, Zv9/danRer7 (Jul, 2010)<br>
		<a target='_blank' href='ftp://hgdownload.soe.ucsc.edu/goldenPath/danRer7/bigZips/danRer7.fa.gz'>
		ftp://hgdownload.soe.ucsc.edu/goldenPath/danRer7/bigZips/danRer7.fa.gz</a>
	<li><b>Sea squirt (<i>Ciona intestinalis</i>) genome</b>, JGI 2.1/ci2 (Mar, 2005)<br>
		<a target='_blank' href='ftp://hgdownload.soe.ucsc.edu/goldenPath/ci2/bigZips/ScaffoldFa.zip'>
		ftp://hgdownload.soe.ucsc.edu/goldenPath/ci2/bigZips/ScaffoldFa.zip</a>
	<li><b>Fruit fly (<i>Drosophila melanogaster</i>) genome</b>, BDGP R5/dm3 (Apr, 2006)<br>
		<a target='_blank' href='ftp://hgdownload.soe.ucsc.edu/goldenPath/dm3/bigZips/chromFa.tar.gz'>
		ftp://hgdownload.soe.ucsc.edu/goldenPath/dm3/bigZips/chromFa.tar.gz</a>
	<li><b>Roundworm (<i>Caenorhabditis elegans</i>) genome</b>, WS220/ce10 (Oct, 2010)<br>
		<a target='_blank' href='ftp://hgdownload.soe.ucsc.edu/goldenPath/ce10/bigZips/chromFa.tar.gz'>
		ftp://hgdownload.soe.ucsc.edu/goldenPath/ce10/bigZips/chromFa.tar.gz</a>
	<li><b>Thale cress (<i>Arabidopsis thaliana</i>) genome</b>, TAIR10 (Nov, 2010)<br>
		<a target='_blank' href='ftp://ftp.arabidopsis.org/home/tair/Sequences/whole_chromosomes/'>
		ftp://ftp.arabidopsis.org/home/tair/Sequences/whole_chromosomes</a>/*.fas
	<li><b>Rice (<i>Oryza sativa</i>) genome</b>, Os-Nipponbare-Reference-IRGSP-1.0 (Oct, 2011)<br>
		Genome assemblies (12 chromosomes) + Unanchored sequences from<br>
		<a target='_blank' href='http://rapdb.dna.affrc.go.jp/download/irgsp1.html'>
		http://rapdb.dna.affrc.go.jp/download/irgsp1.html</a>
	<li><b>Sorghum (<i>Sorghum bicolor</i>) genome</b>, Sorghum bicolor v2.1 (May, 2013)<br>
		<a target='_blank' href='ftp://ftp.jgi-psf.org/pub/compgen/phytozome/v9.0/early_release/Sbicolor_v2.1/assembly/Sbicolor_v2.1_255.fa.gz'>
		ftp://ftp.jgi-psf.org/pub/compgen/phytozome/v9.0/early_release/Sbicolor_v2.1/assembly/Sbicolor_v2.1_255.fa.gz</a>
	<li><b>Silkworm (<i>Bombyx mori</i>) genome</b>, Bmor1 (Apr, 2008)<br>
		Bombyx_mori.Bmor1.17.dna.toplevel.fa.gz from<br>
		<a target='_blank' href='ftp://ftp.ensemblgenomes.org/pub/metazoa/release-17/fasta/bombyx_mori/dna/'>
		ftp://ftp.ensemblgenomes.org/pub/metazoa/release-17/fasta/bombyx_mori/dna/</a>
	<li><b>Budding yeast (<i>Saccharomyces cerevisiae</i>) (S288C) genome</b>, sacCer3 (Apr, 2011)<br>
		<a target='_blank' href='ftp://hgdownload.soe.ucsc.edu/goldenPath/sacCer3/bigZips/chromFa.tar.gz'>
		ftp://hgdownload.soe.ucsc.edu/goldenPath/sacCer3/bigZips/chromFa.tar.gz</a>
	<li><b>Fission yeast (<i>Schizosaccharomyces pombe</i>) (972h-) genome</b>, ASM294v2 (Nov, 2007)<br>
		<a target='_blank' href='ftp://ftp.ensemblgenomes.org/pub/fungi/current/fasta/schizosaccharomyces_pombe/dna/Schizosaccharomyces_pombe.ASM294v2.23.dna.genome.fa.gz'>
		ftp://ftp.ensemblgenomes.org/pub/fungi/current/fasta/schizosaccharomyces_pombe/dna/Schizosaccharomyces_pombe.ASM294v2.23.dna.genome.fa.gz</a>
</ul>

<h5>Illustrated overview of specificity check</h5>

<img src='gRNAalign.png' width=557 height=422 border=0>


<h4>API</h4>

<p>CRISPRdirect results can be directly retrieved in HTML, tab-delimited text or
	JSON format by simply POSTing these values to http://crispr.dbcls.jp/ using
	any kind of language or script. A sample Perl code is available from
	<a target='_blank' href='https://github.com/meso-cacase/CRISPRdirect_crawl'>GitHub</a>.</p>

<table cellpadding=0 cellspacing=0 id='apiman'>
<tr>
	<th>Name</th>
	<th>Required<br>or Optional</th>
	<th>Description</th>
	<th>Sample Value</th>
</tr>
<tr>
	<td><b><i>userseq</i></b></td>
	<td>Required</td>
	<td>A nucleotide sequence; FASTA format or<br>
		a plain nucleotide sequence up to 10 kbp.</td>
	<td>atgccgcgcgtcgtgcccgaccagagaagc ...</td>
</tr>
<tr>
	<td><b><i>accession</i></b></td>
	<td>Optional</td>
	<td>Retrieve sequence mode. Set an accession<br>
		number to retrieve sequence from GenBank<br>
		instead of designing CRISPR/Cas targets.<br>
		Genome location is also accepted for<br>
		hg19, mm10, rn5, galGal4, xenTro3,<br>
		danRer7, ci2, dm3, ce10 and sacCer3.<br>
		The variable <i>userseq</i> should be null.<br>
		The variable <i>format</i> should be 'html' or 'txt'.</td>
	<td>NM_001187<br>
		hg19:chr7:900000-901000</td>
</tr>
<tr>
	<td><b><i>pam</i></b></td>
	<td>Optional</td>
	<td>An arbitrary 3 nt sequence using IUB codes<br>
		can be specified.</td>
	<td>NGG (default), NAG, ...</td>
</tr>
<tr>
	<td><b><i>db</i></b></td>
	<td>Optional</td>
	<td>Set species for off-target searching.</td>
	<td>hg19    (Human         [<i>Homo sapiens</i>]; default),  <br>
	    mm10    (Mouse         [<i>Mus musculus</i>]),           <br>
	    rn5     (Rat           [<i>Rattus norvegicus</i>]),      <br>
	    calJac3 (Marmoset      [<i>Callithrix jacchus</i>]),     <br>
	    susScr3 (Pig           [<i>Sus scrofa</i>]),             <br>
	    galGal4 (Chicken       [<i>Gallus gallus</i>]),          <br>
	    xenTro3 (Frog          [<i>Xenopus tropicalis</i>]),     <br>
	    Xenla7  (Frog          [<i>Xenopus laevis</i>]),         <br>
	    danRer7 (Zebrafish     [<i>Danio rerio</i>]),            <br>
	    ci2     (Sea squirt    [<i>Ciona intestinalis</i>]),     <br>
	    dm3     (Fruit fly     [<i>Drosophila melanogaster</i>]),<br>
	    ce10    (Roundworm     [<i>Caenorhabditis elegans</i>]), <br>
	    TAIR10  (Thale cress   [<i>Arabidopsis thaliana</i>]),   <br>
	    rice    (Rice          [<i>Oryza sativa</i>]),           <br>
	    sorBic  (Sorghum       [<i>Sorghum bicolor</i>]),        <br>
	    bmor1   (Silkworm      [<i>Bombyx mori</i>]),            <br>
	    sacCer3 (Budding yeast [<i>Saccharomyces cerevisiae</i>])<br>
	    pombe   (Fission yeast [<i>Schizosaccharomyces pombe</i>])</td>
</tr>
<tr>
	<td><b><i>format</i></b></td>
	<td>Optional</td>
	<td>Output format.</td>
	<td>html (default), txt, json</td>
</tr>
<tr>
	<td><b><i>download</i></b></td>
	<td>Optional</td>
	<td>Set 'download' to download result as a file.</td>
	<td>download</td>
</tr>
</table>


<h4>Useful tools</h4>

<ul>
	<li><b>GGGenome</b>: ultrafast genome sequence search engine maintained by our group.<br>
		<a target='_blank' href='http://GGGenome.dbcls.jp/'>http://GGGenome.dbcls.jp/</a>
	<li><b>siDirect</b>: highly effective, target-specific siRNA design software maintained by our group.<br>
		<a target='_blank' href='http://siDirect2.RNAi.jp/'>http://siDirect2.RNAi.jp/</a>
</ul>


<h4>Reference</h4>

<p>If you use CRISPRdirect in your work, please cite:</p>

<ul>
	<li>Naito Y, Hino K, Bono H, Ui-Tei K. (2015)<br>
		<a href='http://dx.doi.org/10.1093/bioinformatics/btu743'>
		CRISPRdirect: software for designing CRISPR/Cas guide RNA with reduced off-target sites.</a><br>
		<i>Bioinformatics</i>, <b>31</b>, 1120-1123.
</ul>


<h4 class=s>CRISPRdirect team</h4>

<p class=s>
	Yuki Naito<sup>1</sup>,
	Kimihiro Hino<sup>2</sup>,
	Hidemasa Bono<sup>1</sup>,
	Kumiko Ui-Tei<sup>2</sup><br>
<sup>1</sup>Database Center for Life Science
	(<a target='_blank' href='http://dbcls.rois.ac.jp/en/'>DBCLS</a>) and
<sup>2</sup>University of Tokyo
</p>

<hr style='border-top:1px solid #2655A9'><!-- ________________________________________ -->

<p class=s>
	<a href='..'>CRISPRdirect</a> |
	<a href='http://dbcls.rois.ac.jp/en/'>DBCLS</a><br>
	This page is licensed under a
	<a target='_blank' class=a href='http://creativecommons.org/licenses/by/2.1/jp/deed.en'>
	Creative Commons Attribution 2.1 Japan License</a>.
</p>

<p class=s>
	If you use CRISPRdirect in your work, please cite:<br>
	Naito Y, Hino K, Bono H, Ui-Tei K. (2015)<br>
	CRISPRdirect: software for designing CRISPR/Cas guide RNA with reduced off-target sites.<br>
	<i>Bioinformatics</i>, <b>31</b>, 1120-1123.
	<a target='_blank'
		href='http://bioinformatics.oxfordjournals.org/content/31/7/1120.full'>
		[Full Text]
	</a>
</p>

</body>
</html>
