<!DOCTYPE HTML PUBLIC '-//W3C//DTD HTML 4.01 Transitional//EN'>
<html>

<head>
<meta http-equiv='Content-Type' content='text/html; charset=utf-8'>
<meta http-equiv='Content-Style-Type' content='text/css'>
<meta name='author' content='Yuki Naito'>
<title>CRISPRdirect</title>
<style type='text/css'>
<!--
	*     { font-family:verdana,arial,helvetica,sans-serif }
	.mono { font-family:monaco,courier,monospace }
	p,div,ol,ul,h4 { font-size:9pt }
	table,textarea { font-size:8pt }
	a       { text-decoration:none; color:#003399 }
	a:hover { text-decoration:none; color:maroon  }
	.b      { text-decoration:none; color:black   }
	table { text-align:center; white-space:nowrap }
	th { border-left:1px solid white }
	td { border-left:1px solid #CCCCCC }
	.v { background-color:#E6E6FA }
	.o { background-color:#FFF0E0 }
	.g { background-color:#E6FAF2 }
	.p { background-color:#FFEDED }
	.w { background-color:white   }
-->
</style>
</head>

<body>

<div>
<a href='..'><img src='CRISPRdirect.png' alt='CRISPRdirect' width=200 height=105 border=0></a>
<span style='vertical-align:28px'>&mdash; Rational design of CRISPR/Cas target.</span>
</div>

<hr style='border-top:3px solid #2655A9'><!-- ________________________________________ -->


<h3>About CRISPRdirect</h3>

<p>CRISPRdirect is a web server for selecting rational CRISPR/Cas targets within an input sequence.


<h3>Example</h3>

<h4>Selecting target sites for human B melanoma antigen (BAGE) gene</h4>

<ol>
	<li>Enter the accession number for human BAGE gene (NM_001187).
	<li>Click 'retrieve sequence' to get nucleotide sequence from GenBank.
	<li>Or you can directly paste in a nucleotide sequence.<br>
		Accepted input types are FASTA or plain nucleotide sequence up to 10 kbp.
	<li>Select 'Human genome, GRCh37/hg19 (Feb, 2009)' for off-target checking.
	<li>Click 'design'.
</ol>

<img src='CRISPR_BAGE_fasta.png' width=525 height=404 border=0>

<h4>Results</h4>

<ol type=A>
	<li>Target positions.
	<li>Target sequences, 20mer + 3mer PAM (total 23mer).
	<li>GC content of the target 20mer.
	<li>Calculated Tm of the target 20mer.
	<li>Presence or absence of TTTT in the target sequence. Avoid TTTT for gRNA vectors with pol III promoter.
	<li>Off-target search results against human genome. Number of hits with perfect match is shown.
	<ul>
		<li>20mer+PAM: The target 20mer is searched against the genome.<br>
			The number '1' in this column usually indicates a perfect match against intended target (<i>e.g.</i>, BAGE gene).<br>
			The number '0' in this column means no match in the genome. Possibly the sequence over exon-exon junction.<br>
			Smaller number of hits (but not zero) is better.
		<li>12mer+PAM: The 12mer of 3&prime; region of the target sequence is searched against the genome.<br>
			This region contains critical residues for target specificity.<br>
			Smaller number of hits (but not zero) is better.
		<li>8mer+PAM: The 8mer of 3&prime; region of the target sequence is searched against the genome.<br>
			This region contains the most critical residues for target specificity.<br>
			Smaller number of hits (but not zero) is better.
	</ul>
</ol>

<img src='CRISPR_BAGE_result1.png' width=525 height=306 border=0>

<ol type=A start=7>
	<li>Only one hit in the 20mer and the 12mer columns. These targets are highly specific.<br>
		<font color=brown>&rarr; Recommended for CRISPR/Cas target.</font>
	<li>No hit in the 20mer column. Possibly the sequences over exon-exon junction, so avoid using these ones.<br>
		<font color=brown>&rarr; Not recommended for CRISPR/Cas target.</font>
	<li>Very high number of off-target hits. Avoid using these sequences.<br>
		<font color=brown>&rarr; Not recommended for CRISPR/Cas target.</font>
</ol>

<img src='CRISPR_BAGE_result2.png' width=525 height=272 border=0>

<h4>Saving and exporting the results</h4>

<ol type=A start=10>
	<li>Tab-delimited text can be shown or downloaded as a file.
	<li>JSON format output is also available.
	<li>Tab-delimited text, ready-to-use for copy-pasting into Excel or text editors, etc.
</ol>

<img src='CRISPR_BAGE_tsv.png' width=525 height=453 border=0>


<h3>Methods</h3>

<h4>Specificity check</h4>

<p>Number of off-target hits are displayed for following species:</p>

<ul>
	<li><b>Human genome</b>, GRCh37/hg19 (Feb, 2009)<br>
		<a target='_blank' href='ftp://hgdownload.soe.ucsc.edu/goldenPath/hg19/bigZips/chromFa.tar.gz'>
		ftp://hgdownload.soe.ucsc.edu/goldenPath/hg19/bigZips/chromFa.tar.gz</a>
	<li><b>Mouse genome</b>, GRCm38/mm10 (Dec, 2011)<br>
		<a target='_blank' href='ftp://hgdownload.soe.ucsc.edu/goldenPath/mm10/bigZips/chromFa.tar.gz'>
		ftp://hgdownload.soe.ucsc.edu/goldenPath/mm10/bigZips/chromFa.tar.gz</a>
	<li><b>Rat genome</b>, RGSC 5.0/rn5 (Mar, 2012)<br>
		<a target='_blank' href='ftp://hgdownload.soe.ucsc.edu/goldenPath/rn5/bigZips/chromFa.tar.gz'>
		ftp://hgdownload.soe.ucsc.edu/goldenPath/rn5/bigZips/chromFa.tar.gz</a>
	<li><b>Chicken genome</b>, ICGSC Gallus_gallus-4.0/galGal4 (Nov, 2011)<br>
		<a target='_blank' href='ftp://hgdownload.soe.ucsc.edu/goldenPath/galGal4/bigZips/galGal4.fa.gz'>
		ftp://hgdownload.soe.ucsc.edu/goldenPath/galGal4/bigZips/galGal4.fa.gz</a>
	<li><b><i>Xenopus tropicalis</i> genome</b>, JGI 4.2/xenTro3 (Nov, 2009)<br>
		<a target='_blank' href='ftp://hgdownload.soe.ucsc.edu/goldenPath/xenTro3/bigZips/xenTro3.fa.gz'>
		ftp://hgdownload.soe.ucsc.edu/goldenPath/xenTro3/bigZips/xenTro3.fa.gz</a>
	<li><b>Zebrafish genome</b>, Zv9/danRer7 (Jul, 2010)<br>
		<a target='_blank' href='ftp://hgdownload.soe.ucsc.edu/goldenPath/danRer7/bigZips/danRer7.fa.gz'>
		ftp://hgdownload.soe.ucsc.edu/goldenPath/danRer7/bigZips/danRer7.fa.gz</a>
	<li><b><i>Ciona intestinalis</i> genome</b>, JGI 2.1/ci2 (Mar, 2005)<br>
		<a target='_blank' href='ftp://hgdownload.soe.ucsc.edu/goldenPath/ci2/bigZips/ScaffoldFa.zip'>
		ftp://hgdownload.soe.ucsc.edu/goldenPath/ci2/bigZips/ScaffoldFa.zip</a>
	<li><b><i>Drosophila</i> genome</b>, BDGP R5/dm3 (Apr, 2006)<br>
		<a target='_blank' href='ftp://hgdownload.soe.ucsc.edu/goldenPath/dm3/bigZips/chromFa.tar.gz'>
		ftp://hgdownload.soe.ucsc.edu/goldenPath/dm3/bigZips/chromFa.tar.gz</a>
	<li><b><i>C. elegans</i> genome</b>, WS220/ce10 (Oct, 2010)<br>
		<a target='_blank' href='ftp://hgdownload.soe.ucsc.edu/goldenPath/ce10/bigZips/chromFa.tar.gz'>
		ftp://hgdownload.soe.ucsc.edu/goldenPath/ce10/bigZips/chromFa.tar.gz</a>
	<li><b><i>Arabidopsis thaliana</i> genome</b>, TAIR10 (Nov, 2010)<br>
		<a target='_blank' href='ftp://ftp.arabidopsis.org/home/tair/Sequences/whole_chromosomes/'>
		ftp://ftp.arabidopsis.org/home/tair/Sequences/whole_chromosomes</a>/*.fas
	<li><b>Rice genome</b>, Os-Nipponbare-Reference-IRGSP-1.0 (Oct, 2011)<br>
		Genome assemblies (12 chromosomes) + Unanchored sequences from<br>
		<a target='_blank' href='http://rapdb.dna.affrc.go.jp/download/irgsp1.html'>
		http://rapdb.dna.affrc.go.jp/download/irgsp1.html</a>
	<li><b>Silkworm genome</b>, Bmor1 (Apr, 2008)<br>
		Bombyx_mori.Bmor1.17.dna.toplevel.fa.gz from<br>
		<a target='_blank' href='ftp://ftp.ensemblgenomes.org/pub/metazoa/release-17/fasta/bombyx_mori/dna/'>
		ftp://ftp.ensemblgenomes.org/pub/metazoa/release-17/fasta/bombyx_mori/dna/</a>
	<li><b><i>S. cerevisiae</i> (S288C) genome</b>, sacCer3 (Apr, 2011)<br>
		<a target='_blank' href='ftp://hgdownload.soe.ucsc.edu/goldenPath/sacCer3/bigZips/chromFa.tar.gz'>
		ftp://hgdownload.soe.ucsc.edu/goldenPath/sacCer3/bigZips/chromFa.tar.gz</a>
</ul>


<h3>Useful tools</h3>

<ul>
	<li><b>GGGenome</b>: ultrafast genome sequence search engine maintained by our group.<br>
		<a target='_blank' href='http://GGGenome.dbcls.jp/'>http://GGGenome.dbcls.jp/</a>
	<li><b>siDirect</b>: highly effective, target-specific siRNA design software maintained by our group.<br>
		<a target='_blank' href='http://siDirect2.RNAi.jp/'>http://siDirect2.RNAi.jp/</a>
</ul>


<hr style='border-top:1px solid #2655A9'><!-- ________________________________________ -->

<p><font color=gray>CRISPRdirect |
<a target='_blank' href='http://twitter.com/meso_cacase'>@meso_cacase</a> at 
<a target='_blank' href='http://dbcls.rois.ac.jp/'>DBCLS</a>
</font></p>

</body>
</html>
